lib/Markdown/Perl/BlockParser.pm

```diff
@@ -372,11 +379,14 @@ sub _parse_yaml_metadata {
     my $metadata = eval { YAML::Tiny->read_string($+{YAML}) };
     if ($EVAL_ERROR) {
       pos($this->{md}) = 0;
-      return;
+      carp 'YAML Metadata (Markdown frontmatter) is invalid.';
+      return 1;
+    }
+    if(exists($this->{pmarkdown}->{hooks}->{yaml_metadata})) {
```

Comment:
nit: you can omit the -> in between sets of {} to follow the same style as the rest of the code.